### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by lalo**
For context this file will be consumed by github in certain parts of the UI, for more info: https://docs.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Once the comments in the pull request have been addressed, a project
+maintainer will merge your changes. Thank you for helping improve Vowpal Wabbit!
+
+By default we'll perform a squash merge unless requested otherwise in the
+PR.
+
+# Reporting Security Issues
+
+Security issues and bugs should be reported privately, via email, to *insert
+email*
```
Comment (lalo) [inline CONTRIBUTING.md:93]: This is optional, but good to have. We just need to have an email.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+have a brief discussion about the problem and, hopefully, identify some
+potential pitfalls before too much time is spent on your part.
+
+### Implementation
+
+* Fork the repository on GitHub.
+* Start on a new topic branch off of master.
+* Instructions for getting Vowpal Wabbit building and running the tests are in the
+  [Wiki](https://github.com/VowpalWabbit/vowpal_wabbit/wiki).
+* Aim for each pull request to have one goal. If the commit starts to get
```
Comment (lalo) [inline CONTRIBUTING.md:31]: replace with: if the PR starts

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+[main repository's master branch](https://github.com/VowpalWabbit/vowpal_wabbit/tree/master).
+
+When you submit a pull request, a suite of builds and tests will be run
+automatically, and the results will show up in the "Checks" section of the
+PR. If any of these fail, you'll need to figure out why and make the
+appropriate fixes. If you think the failures are due to infrastructure
+issues, please mention this in a comment, and one of the maintainers will
+help.
+
+The project maintainers will review your changes. We aim to review all
```
Comment (lalo) [inline CONTRIBUTING.md:74]: We could add something like we prioritize PRs with green runs and that's when the 3 days kick-in.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+Vowpal Wabbit welcomes contributions from the community. Consider this document
+as a guideline rather than strict rules. These are just a set of suggestions
+that we find useful to increase the productivitity of everyone involved.
+
+## Process
+
+1. Make a proposal.
+1. Implement the proposal and its tests.
+1. Squash commits and write a good commit message.
```
Comment (lalo) [inline CONTRIBUTING.md:11]: not necessary to say squash, just write a good commit message

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# Contributing
+
+Vowpal Wabbit welcomes contributions from the community. Consider this document
+as a guideline rather than strict rules. These are just a set of suggestions
+that we find useful to increase the productivitity of everyone involved.
+
+## Process
+
+1. Make a proposal.
```
Comment (jackgerrits) [inline CONTRIBUTING.md:9]: Nit: unless it is a full sentence a period isn't needed for a list item

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+1. [Write a good commit message](#suggested-commit-structure)
+1. [Start a pull request & address comments](#pull-request)
+1. [Merge](#merge)
+
+### Proposal
+
+For things like fixing typos and small bug fixes, you can skip this step.
+
+If your change is more than a simple fix, please don't just create a big
+pull request. Instead, start by opening an issue describing the problem you
```
Comment (lalo) [inline CONTRIBUTING.md:20]: problem/feature

**Comment by JohnLangford** [comment]
Reading through the text, some thoughts.
(1) "large" and "small" are sort of unquantified.  Maybe this is unavoidable?  But I could imagine people being unclear.
(2) It's not super-clear to me that we need a proposal from someone doing something modular (like a reduction) or someone who is experienced.  Avoiding a barrier to entry seems good.
(3) It's unclear what's being proposed between "1 patch, 1 goal" and "if it's to large, break it up."
(4) The formatting for the pull request commit seems plausibly over-specific. What if it's not in the imperative?

**Comment by JohnLangford** [comment]
The current draft reads well to me.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+Once the comments in the pull request have been addressed, we will merge your
+changes. Thank you for helping improve Vowpal Wabbit!
+
+By default we'll perform a squash merge, so don't worry if the commit history is
+messy.
+
+# Reporting Security Issues
+
+Security issues and bugs should be reported privately, via email, to `vowpalwabbit-security@hunch.net`
```
Comment (lalo) [inline CONTRIBUTING.md:61]: should we seperate the email to avoid getting it parsed by all the spam bots?

vowpalwabbit-security [a t] hunch.net

Comment (lalo) [reply CONTRIBUTING.md:61]: or something like that

**Review by olgavrou — APPROVED** [review]

File: CONTRIBUTING.md (hunk 7 above)
Comment (jackgerrits) [reply CONTRIBUTING.md]: Done

**Pull request merged**